hwconf/Zerno/hw_zerno_drive_core.c

```diff
@@ -36,727 +36,810 @@
+        offset_value.as_float = encoder_min_value_in_volts;
+        conf_general_store_eeprom_var_hw( &offset_value, EEPROM_ADDR_ENCODER_VALUE );
+        encoder_max_value_in_volts = MAX_ADC_VALUE_IN_VOLTS;
+        is_calibration_done = 1;
+        calibration_check.as_i32 = is_calibration_done;
+        conf_general_store_eeprom_var_hw( &calibration_check, EEPROM_ADDR_CALIBRATION_CHECK );
+        safety_calibration = true;
+    }
 }
 
-void encoder_calibrate_offset(void) {
+void set_erpm_ramp_response( void )
+{
+    mc_configuration * mcconf = mempools_alloc_mcconf();
 
-	eeprom_var offset_value, calibration_check;
+    *mcconf = *mc_interface_get_configuration();
+    mc_configuration * mcconf_old = mempools_alloc_mcconf();
+    *mcconf_old = *mcconf;
 
-    encoder_total_value = ADC_VOLTS(ADC_IND_EXT); // get the knob position values
-    main_switch_value = ADC_VOLTS(ADC_IND_EXT2); // get the switch position values
+    if( !is_default_erpm )
+    {
+        mcconf->s_pid_ramp_erpms_s = 10000.0;
+    }
+    else
+    {
+        mcconf->s_pid_ramp_erpms_s = 8000.0;
+    }
 
-	if(!is_momentary_position()) { // Digital and analog detection for calibration mode.
-		encoder_min_value = encoder_total_value;
-		offset_value.as_float = encoder_min_value;
-		conf_general_store_eeprom_var_hw(&offset_value, EEPROM_ADDR_ENCODER_VALUE);
-		encoder_max_value = 3.2;
-		is_calibration_done = 1;
-		calibration_check.as_i32 = is_calibration_done;
-		conf_general_store_eeprom_var_hw(&calibration_check, EEPROM_ADDR_CALIBRATION_CHECK);
-		safety_calibration = true;
-	}
+    mc_interface_set_configuration( mcconf_old );
+    mc_interface_set_configuration( mcconf );
+
+    mempools_free_mcconf( mcconf );
+    mempools_free_mcconf( mcconf_old );
+
+    is_erpm_done = true;
 }
 
-void set_erpm_ramp_response(void) {
-	mc_configuration *mcconf = mempools_alloc_mcconf();
-	*mcconf = *mc_interface_get_configuration();
-	mc_configuration *mcconf_old = mempools_alloc_mcconf();
-	*mcconf_old = *mcconf;
+void set_pid_constant( void )
+{
+    mc_configuration * mcconf = mempools_alloc_mcconf();
 
-	if(!is_default_erpm)
-		mcconf->s_pid_ramp_erpms_s = 10000.0;
-	else
-		mcconf->s_pid_ramp_erpms_s = 8000.0;
+    *mcconf = *mc_interface_get_configuration();
+    mc_configuration * mcconf_old = mempools_alloc_mcconf();
+    *mcconf_old = *mcconf;
 
-	mc_interface_set_configuration(mcconf_old);
-	mc_interface_set_configuration(mcconf);
+    if( speed_erpm_setpoint < SPEED_PID_CHANGE )
+    {
+        mcconf->s_pid_kd = 0.000300; // for 7A it is set to kp: 0.000400
+    }
+    else
+    {
+        mcconf->s_pid_kd = 0.000020;
+    }
 
-	mempools_free_mcconf(mcconf);
-	mempools_free_mcconf(mcconf_old);
+    mc_interface_set_configuration( mcconf_old );
+    mc_interface_set_configuration( mcconf );
 
-	is_erpm_done = true;
+    mempools_free_mcconf( mcconf );
+    mempools_free_mcconf( mcconf_old );
 }
 
-void set_pid_constant(void) {
+void encoder_cal_detection( void )
```

Comment:
```suggestion
static void encoder_cal_detection( void )
```

Also, please add `static` when doing the forward declaration.